### PR TITLE
Add support for out of bounds area map entries

### PR DESCRIPTION
**Pull request by krypciak**

### CODE REVIEW

File: src/prestart/menu/custom-map-utils.ts [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+                const newSize: Vec2 = Vec2.create(max)
+                Vec2.sub(newSize, min) /* min has to be negative here */
+                /* create an empty array */
+                const arr: number[][] = []
+                for (let y = 0; y < newSize.y; y++) {
+                    arr[y] = []
+                    for (let x = 0; x < newSize.x; x++) {
+                        arr[y][x] = 0
+                    }
+                }
```
Comment (elluminance) [inline src/prestart/menu/custom-map-utils.ts]: Can be reduced down to either:
```js
for(let y = 0; y < newSize.y; y++) {
    arr[y] = Array(newSize.x).fill(0);
}
```
or even more compactly `const arr = Array.from(Array(newSize.y), () => Array(newSize.x).fill(0))`

File: src/prestart/menu/custom-map-utils.ts [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+                if (map.customMap) {
+                    const c: sc.AreaLoadable.CustomMap = map.customMap
+                    min.x = Math.min(min.x, c.offX)
+                    min.y = Math.min(min.y, c.offY)
+                    max.x = Math.max(max.x, c.offX + c.mapData[0].length)
+                    max.y = Math.max(max.y, c.offY + c.mapData.length)
+                }
+            }
+            if (min.x != 0 || min.y != 0 || max.x != os.x || max.y != os.y) {
+                console.log(floor, min, max)
```
Comment (elluminance) [inline src/prestart/menu/custom-map-utils.ts]: Unnecessary `console.log`?

**Review by elluminance — CHANGES_REQUESTED** [review]
Looks good for the most part! Glad it works as well as it does. I like your clever use of Vec2s, I must say.

However... I did notice one thing when testing an extreme example. If you do a really big offset, the camera won't allow you to scroll and see the whole map. Here's what I used (I had it on a `(-100, -100)` offset. While I don't think most people will go _that_ far - I do think it's possible that larger maps could easily run into this issue (especially if people add maps from both sides). 
[rhombus-sqr.json.patch](https://github.com/EL20202/el-crosscode-tweaks/files/12777296/rhombus-sqr.json.patch)

With that patch above, this is as far as I could scroll to the bottom right.
image

**Comment by krypciak** [comment]
This should work now, the issue you mentioned required changing the area dimensions
Turns out different floor levels weren't aligned when you resize only one of them, so I resize them all

**Review by elluminance — APPROVED** [review]

**Comment by elluminance** [comment]
Looks like that fixed it! Thanks so much - and sorry it took me so long to check this out.

**Pull request merged**